Mod2/Preparation/Week4/LINQ.md

```diff
@@ -0,0 +1,91 @@
+## Pre-work for [LINQ](/Mod2/Lessons/Week4/LINQ.md)
+
+Before our lesson on LINQ, we need to get familiar with some additional C# syntax - `Lambda Expressions`.  Also referred to as `anonymous functions`, lambda expressions are a tool used in many different programming languages.  The goal of this preparation is for you to **see** and **execute** some Lambda expressions - we do not expect you to be a master of Lambda expressions!
+
+Before we define what a lambda expression **is**, let's look at some code examples to see what a lambda expression **does**.
+
+To follow along with the examples below, you can clone [this GitHub Repo](https://github.com/turingschool-examples/Launch_LambdaExpressions)
+
+### Reviewing the Existing Functionality
+
+In the program, we have two methods that are fairly simple - they each perform a caluclation on a number, and return the integer result of that calculation.
+
+```c#
+class Program
+{
+    static void Main(string[] args)
+    {
+        Console.WriteLine(Square(5));
+        Console.WriteLine(Double(5));
+    }
+
+    static int Square(int number)
+    {
+        return number * number;
+    }
+
+    static int Double(int number)
+    {
+        return number * 2;
+    }
+}
+```
+
+This code works just fine, but there are other ways we could accomplish the same goal.  In C# (and many other programming languages), we can refactor very simple methods into `Lambda Expressions`
+
+### Refactored into Lambda
+
+```c#
+    class Program
+    {
+        static void Main(string[] args)
+        {
+            Func<int, int> Square = number => number * number;
+            Func<int, int> Double = number => number * 2;
+
+            Console.WriteLine(Square(5));
+            Console.WriteLine(Double(5));
+        }
+    }
+```
+
+This probably looks a little weird, but if we dig into what's happening, we can draw some similarities between the `Square` and `Double` lambda expressions and methods.  Like methods, lambda expressions:
+* Take input that must be of a specified type.
+* Output a return value that must be of a specified type.
+* Have a name that we use to execute the expression.
+
+![](/Mod2/Images/Week4/LambdaExpressionDiagram.png)
+
+### The Power of Lambda
+
+One benefit of a lambda expression is that they can be used like objects in our programs.  Specifically, they can be passed as arguments into certain methods.
+
+Consider an alternative refactor to our program.  Both `Square` and `Double` take an int as an argument, and perform some operation on that number, and return an int.  Because lambda expressions are a type of object, we could create a method that can process a given operation on a given number:
+
+```c#
+    class Program
+    {
+        static void Main(string[] args)
+        {
+            Console.WriteLine($"Square: {Calculate(5, number => number * number)}");
+            Console.WriteLine($"Double: {Calculate(5, number => number * 2)}");
```

Comment:
I think this is a good example to show how you can pass in a function to another funciton, but also there is a lot happening in this chunk of code. I wonder if it would be clearer to have the Calculate function Console.WriteLine the result of the function passed in instead of return it. I think that would make the main a bit easier to read?

I would also maybe show an example of using the same `number => number * number` but passing in different values?

Reply:
Maybe we encourage students to comment out the different lines in the main to see what they output? Just trying to think of how it might be easier to see this part
```calculate(5, number => number * number)```

Reply:
I pulled the Calculate method call out of the WriteLine - it is a lot of noise there.